### ISSUE TITLE
[Assistive tech and device support] Keyboard navigation order isn't meaningful or sequential. (11.08.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#34872]
#### VFS team name
Claims & Appeals

#### VFS product name
HLR V2

#### Point of Contact/Reviewers
Angela Fowler (@AngelaFowler82) - Accessibility

---

### Platform Issue
Keyboard navigation order isn't meaningful or sequential.

### Issue Details
When filling out the application, the progress indicater can be accessed using the tab key. This could fool keyboard users into thinking it is clickable.

### VA.gov Experience Standard
[Category Number 11, Issue Number 08](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 2.4.3_A

---

### Platform Recommendation
Remove the progress indicater from the tab order.

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] Close the ticket when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment in the ticket.

**Comment by shiragoodman** [comment]
Follow up on this as it may be a progress bar issue.

**Comment by Mottie** [comment]
I have reported this issue to the design system team and plan on following up on how to address this problem.

**Comment by Mottie** [comment]
Related to https://github.com/department-of-veterans-affairs/va.gov-team/issues/21138
CC: @caw310

**Comment by Mottie** [comment]
Yes, reassign please

**Comment by shiragoodman** [comment]
@caw310 @k80bowman This ticket has been reassigned from Claims and Appeals team to Design System team. Please let me know if you have any questions.

**Comment by taylorkaren** [comment]
Moved to ticket https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/249, closing this copy of the issue.

**Comment by shiragoodman** [comment]
@taylorkaren please do not close the ticket if the issue is not resolved. This ticket is used to help populated our Collab Cycle Dashboard in domo. Closing it will give us a false data set.

**Comment by bjmccotter7192** [comment]
After looking into the Forms System, it's possible that this needs to be updated in the va-segmented-progress-bar web-component and the tabIndex should be updated there so that keyboard navigation is not triggered.

Tried to bypass it by adding to the outer div around the web-component but doesn't seem to do anything.

@k80bowman @bkjohnson 

https://github.com/department-of-veterans-affairs/component-library/blob/master/packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx#L66

**Comment by bkjohnson** [comment]
There's nothing in [the component guidance](https://design.va.gov/components/progress-bar/segmented) saying that it _should_ be focusable, so maybe removing that `tabindex="0"` is the right thing. I traced the history [back to 2017](https://github.com/department-of-veterans-affairs/design-system/pull/8/files#diff-61a92cf9e4794845b24146ffe44c42fe03129aec5acada59b9e8ce6d6546b37dR13) and the PR doesn't say anything about `tabindex="0"` solving any UX or a11y problems so it seems safe to remove, but we should probably get an accessibility expert to confirm that.

**Comment by taylorkaren** [comment]
There is a hard coded tab index in the component, so bypassing it doesn't do any good. Remove the tab index or change it to -1 so it is not navigable by keyboard but would be navigable by JS. BJ can put a PR in for the Design System Team to review and have an accessibility specialist review the change.

**Comment by joshkimux** [comment]
Small notes and changes here 🙇 
- This doesn't completely block an assistive tech user from completing their tasks, so I'm removing the 508-defect-1 and launchblocking label in favor of the 508-defect-2 label
- Can confirm removing the progress indicator from the tab order would be great (which I think we wrote a ticket for before 🤔 )
- This may not be relevant pending future changes to the form system (e.g. will the progress indicator even still exist in the format and styling it is now?)

**Comment by bkjohnson** [comment]
@joshkimux, is removing it from the tab order recommended, or should it have `tabindex="-1"` for any kind of programmatic focus?

**Comment by joshkimux** [comment]
@bkjohnson given it isn't interactive and that there are written steps, I don't see the need of it ever needing programmatic focus. Removing the `tabindex` completely should do the trick 😄

**Comment by taylorkaren** [comment]
@joshkimux: This work has been completed and merged via ticket https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/249. If that satisfies what you were looking for, please close this ticket.

**Comment by joshkimux** [comment]
Thank you @taylorkaren and team 🥳

**Comment by joshkimux** [comment]
sitewide! Adding that on now :)
